### PR TITLE
Fix `displayText()` for null values in structure field

### LINKED ISSUE #4096: Table block preview uses too much padding
## Description

The rows and / or cells of the table block preview use too much padding. The less horizontal space the block has to use, the more obvious it gets.

**Expected behavior**  

The rows and / or cells to use less padding.

**Screenshots**  

## To reproduce

1. Create a table block blueprint (e. g. the [one from the docs](https://getkirby.com/docs/reference/panel/blocks/table)).
2. Use it in a `blocks` or `layout` field.
3. Add at least one row.

## Your setup
**Kirby Version**  

3.6.1.1

**Your system (please complete the following information)**
 - Device: MacBook Pro
 - OS: macOS 12.1
 - Browser: Google Chrome
 - Version: 97.0.4692.71

### LINKED ISSUE #4164: Table Block throws error when value is empty
## Description

If a value is added in a table block and left empty the table fails to render in Vue due to an `Cannot read properties of undefined` error.

If the file is saved and the page is reloaded the table renders as expected.

**Expected behavior**  

An empty value should render the cell with `–` as shown in the screenshot below.

**Screenshots**  

Bildschirmfoto 2022-02-17 um 15 21 23
Bildschirmfoto 2022-02-17 um 15 26 17

## To reproduce

1. Create Table Block blueprint with rows (Types could be text, number etc.)
2. Insert one row with an empty value
3. Save block

## Your setup
**Kirby Version**  
3.6.2

**Console output**  
```
index.js:1 TypeError: Cannot read properties of undefined (reading 'toString')
    at a.displayText (index.js:1:177608)
    at index.js:1:293333
    at a.be [as _l] (vendor.js:6:16323)
    at index.js:1:293013
    at a.be [as _l] (vendor.js:6:16023)
    at a.<anonymous> (index.js:1:292964)
    at a.t._render (vendor.js:6:35359)
    at a.r (vendor.js:6:68538)
    at vn.get (vendor.js:6:26734)
    at vn.run (vendor.js:6:27621)
```
**Your system (please complete the following information)**
 - Device: MacBook Pro
 - OS: macOS
 - Browser: Chrome
 - Version: 98

**Additional context**

### PR DESCRIPTION
**Pull request by afbora**
## Describe the PR

Optional chaining is used to avoid error when value is `null` or `undefined`. The return value has been verified to work where `displayText()` is used.

### Fixes

- Fixed table block throws error when value is empty #4164

## Breaking changes

None

## Related issues/ideas

- Fixes #4164

## Ready?

- [ ] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] CI checks pass

## When merging

- [x] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)
- [ ] Add changes to release notes draft in Notion

### CODE REVIEW

File: panel/src/components/Forms/Blocks/Types/Table.vue [hunk 1]
```diff
@@ -24,19 +24,21 @@
...
-          :column="column"
-          :field="fields[columnName]"
-        />
-        <template v-else>
-          <p class="k-structure-table-text">
-            {{ column.before }}
-            {{ displayText(fields[columnName], row[columnName]) || "–" }}
-            {{ column.after }}
-          </p>
+        <template v-if="columnIsEmpty(row[columnName]) === false">
```
Comment (afbora) [inline panel/src/components/Forms/Blocks/Types/Table.vue]: To only show when value is not empty, like we do in structure field

File: panel/src/components/Forms/Blocks/Types/Table.vue [hunk 2]
```diff
@@ -89,14 +91,24 @@ export default {
   border-radius: var(--rounded-sm);
   overflow: hidden;
   table-layout: fixed;
+  --item-height: 38px;
```
Comment (afbora) [inline panel/src/components/Forms/Blocks/Types/Table.vue]: To stable height when all row values are empty

File: panel/src/components/Forms/Blocks/Types/Table.vue [hunk 3]
```diff
@@ -89,14 +91,24 @@ export default {
...
+.k-block-type-table-preview th {
+  padding: 0.5rem 0.75rem;
+}
+.k-block-type-table-preview td {
+  height: var(--item-height);
+  padding: 0 0.75rem;
+}
+.k-block-type-table-preview td > * {
+  padding: 0;
+}
```
Comment (afbora) [inline panel/src/components/Forms/Blocks/Types/Table.vue]: Prevent extra padding for field previews in cell, for ex: writer field

**Comment by distantnative** [comment]
The one thing I cannot really oversee is the styling changes if they don't have unwanted consequences. The rest looks good to me.

**Comment by afbora** [comment]
@distantnative I've split the PR, see for UI changes https://github.com/getkirby/kirby/pull/4168

**Review by distantnative — APPROVED** [review]

**Pull request merged**